script/test-race-conditions.spec.js

```diff
@@ -0,0 +1,378 @@
+"use strict";
+
+var config, DDQ, done, instance;
+
+config = require("node_modules/ddq/script/manual-testing-config");
+DDQ = require("ddq");
+
+
+/**
+ * Opens a connection to the database and deletes all entries.
+ *
+ * @param {Function} [callback]
+ */
+function cleanDatabase(callback) {
+    /**
+     * Logs errors passed in.
+     *
+     * @param {Error} err
+     */
+    function errorHandler(err) {
+        if (err) {
+            console.log("There was an error during cleanup:", err);
+        }
+    }
+
+    instance.open((err) => {
+        errorHandler(err);
+
+        // This implementation is specific to the MySQL library.
+        instance.backend.connection.query(`
+            DELETE FROM ??`,
+            [
+                instance.backend.config.table
+            ], (queryErr, data) => {
+                errorHandler(queryErr);
+
+                if (data) {
+                    console.log("Records cleaned up:", data.affectedRows);
+                }
+
+                instance.close((endErr) => {
+                    errorHandler(endErr);
+
+                    if (callback) {
+                        callback();
+                    }
+                });
+            }
+        );
+    });
+}
+
+
+/**
+ * Accepts an array of methods and done callback. Each method calls the
+ * following method in it's callback unless there is an error, in which case the
+ * done callback is invoked, breaking the flow.
+ *
+ * @param {array} methods
+ * @param {Function} doneCb
+ */
+function runTests(methods, doneCb) {
+    var nextMethod;
+
+    if (!methods || !methods.length) {
+        doneCb();
+
+        return;
```

Comment:
Do we need a return here if we call the callback? Don't remember how that works.